packages/core/src/StateNode.ts

```diff
@@ -463,6 +267,39 @@ class StateNode<
     return this.definition;
   }
 
+  /**
+   * The services invoked by this state node.
+   */
+  public get invoke(): Array<InvokeDefinition<TContext, TEvent>> {
```

Comment:
Previously this was only evaluated once - in the constructor. You have made it a getter now, but its implementation is impure so it poses a risk of introducing a bug in the future, even if there is none now. At the very least its return should be cached (like it's done for the most of other getters defined on StateNode).

Reply:
Well - it isn't pure, but it's idempotent (sort of), so I guess my remark about it posing a risk is an exaggeration, but probably it's better to cache this anyway.